zerver/actions/realm_settings.py

```diff
@@ -554,3 +555,41 @@ def do_send_realm_reactivation_email(realm: Realm, *, acting_user: Optional[User
         language=language,
         context=context,
     )
+
+
+def do_set_push_notifications_enabled_end_timestamp(
+    realm: Realm, value: Optional[int], *, acting_user: Optional[UserProfile]
+) -> None:
```

Comment:
If we do so, we'd need to add such if statement in tests, test_events etc.. 

Also it'd feel weird to include `push_notifications_enabled_end_timestamp` in `property_types` as `int` when the field is `datetime`.

If we want to do that, we'd need to think of a generic way to support all such cases. Also [this would be updated](https://zulip.readthedocs.io/en/stable/tutorials/new-feature-tutorial.html#update-the-model):
```
When creating a realm property that is not a boolean, Text or integer field, or when adding a field
that is dependent on other fields, do not add the field to the property_types dictionary. The steps
below will point out where to write additional code for these cases.
```

We can do that post 8.0 as that would be an internal change (no API changes). Let me know, I'll file an issue.

Reply:
I guess I'll just leave a comment; I'm not sure this refactor is worth investing in until we've got more timestamp format properties to think about. (Also moved the function to be just after the one it is a variant of).